tests/back/api/test_tasks.py

```diff
@@ -0,0 +1,13 @@
+from DashAI.back.core.config import task_registry
+
+from fastapi.testclient import TestClient
+
+def test_get_all_tasks(client: TestClient):
+    task_names = task_registry.registry.keys()
+    response = client.get("/api/v1/task/")
+    assert response.status_code == 200, response.text
+    data = response.json()
+    for task_name, task_data in data.items():
+        assert task_name in task_names
+        assert task_data["class"] in task_names
+        assert task_name == task_data["class"]
```

Comment:
I am concerned here that the test is not very deterministic as the tasks you load by default in the app may vary (and if all goes well, eventually not load any in the core).

Here I would suggest using a task registry from scratch and adding test tasks by hand:

```suggestion
from DashAI.back.registries.task_registry import TaskRegistry
from DashAI.back.tasks.base_task import BaseTask
from fastapi.testclient import TestClient
from DashAI.back.core import config


class TestTask1(BaseTask):
    ...


class TestTask2(BaseTask):
    ...


test_task_registry = TaskRegistry(initial_components=[TestTask1, TestTask2])
# replace the default tasks with the previously test tasks
config.task_registry._registry = test_task_registry ._registry


def test_get_all_tasks(client: TestClient):
    task_names = task_registry.registry.keys()
    response = client.get("/api/v1/task/")

    assert response.status_code == 200, response.text

    data = response.json()
    assert len(data.keys()) == 2
    for task_name, task_data in data.items():
        assert task_name in task_names
        assert isinstance(task_data, dict)
```

Note that these changes only work if the above corrections are accepted.